app.rb

```diff
@@ -49,53 +49,94 @@ class PixelTrackerApp < Sinatra::Base
     apidoc.to_json
   end
 
-=begin
-  get '/api/v1/pixels/?' do
+  get '/api/v1/campaigns/?' do
     content_type 'application/json'
-    id_list = Pixel.all
+    JSON.pretty_generate(data: Campaign.all)
+  end
+
+  get '/api/v1/campaigns/:id' do
+    content_type 'application/json'
+
+    id = params[:id]
+    campaign = Campaign[id]
+    trackers = campaign ? Campaign[id].trackers : []
+
+    if campaign
+      JSON.pretty_generate(data: campaign, relationships: trackers)
+    else
+      halt 404, "PROJECT NOT FOUND: #{id}"
+    end
+  end
+
+  post '/api/v1/campaigns/?' do
+    begin
+      new_data = JSON.parse(request.body.read)
+      saved_campaign = Campaign.create(new_data)
+    rescue => e
+      logger.info "FAILED to create new campaign: #{e.inspect}"
+      halt 400
+    end
+
+    new_location = URI.join(@request_url.to_s + '/', saved_campaign.id.to_s).to_s
 
-    { pixel_id: id_list }.to_json
+    status 201
+    headers('Location' => new_location)
   end
 
-  get '/api/v1/pixels/:id.json' do
+  get '/api/v1/campaigns/:id/trackers/?' do
+    content_type 'application/json'
+
+    campaign = Campaign[params[:id]]
+
+    JSON.pretty_generate(data: campaign.trackers)
+  end
+
+  get '/api/v1/campaigns/:campaign_id/trackers/:id/?' do
     content_type 'application/json'
 
     begin
-      { pixel: Pixel.find(params[:id]) }.to_json
+      doc_url = URI.join(@request_url.to_s + '/', 'document')
+      tracker = Tracker
+                .where(campaign_id: params[:campaign_id], id: params[:id])
+                .first
+      halt(404, 'Tracker not found') unless tracker
+      JSON.pretty_generate(data: {
+                             tracker: tracker,
+                             links: { document: doc_url }
+                           })
     rescue => e
-      status 404
-      logger.info "FAILED to GET pixel: #{e.inspect}"
+      status 400
+      logger.info "FAILED to process GET tracker request: #{e.inspect}"
+      e.inspect
     end
   end
 
-  get '/api/v1/pixels/:id/views' do
+  get '/api/v1/campaigns/:campaign_id/trackers/:id/document' do
```

Comment:
I think we don't need to return documents, in our case JSON makes more sense at get '/api/v1/campaigns/:campaign_id/trackers/:id/?'